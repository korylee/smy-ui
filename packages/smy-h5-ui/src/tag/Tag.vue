<template>
  <div :class="bem({ [type]: type, plain, round, mark })" :style="style" @click="$emit('click', $event)">
    <slot></slot>
    <smy-icon v-if="closeable" :class="bem('close')" size="12" @click.stop="$emit('close', $event)"
      ><slot name="icon"><window-close /></slot
    ></smy-icon>
  </div>
</template>

<script>
import { props } from './props'
import Icon from '../icon'
import WindowClose from '@smy-h5/icons/dist/es/WindowClose'
import { createNamespace } from '../_utils/vue/create'

const [name, bem] = createNamespace('tag')

export default {
  name,
  components: { [Icon.name]: Icon, WindowClose },
  props,
  computed: {
    style() {
      const style = {}
      const { textColor, color, plain } = this
      if (textColor) {
        style.color = textColor
      } else if (color && plain) {
        style.color = color
      }
      if (plain) {
        style.background = '#fff'
        style.borderColor = color
      } else if (color) {
        style.background = color
      }
      return style
    },
  },
  methods: {
    bem,
  },
}
</script>

<style lang="less">
@import './tag.less';
</style>
