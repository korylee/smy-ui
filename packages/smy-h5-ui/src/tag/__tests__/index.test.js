import { mount } from '@vue/test-utils'
import Tag from '..'

test('should emit close event when clicking the close icon', () => {
  const wrapper = mount(Tag, {
    propsData: {
      closeable: true,
    },
  })

  wrapper.find('.smy-tag__close').trigger('click')
  expect(wrapper.emitted('close').length).toEqual(1)

  wrapper.destroy()
})

test('should render textColor correctly', () => {
  const wrapper = mount(Tag, {
    propsData: {
      plain: true,
      color: 'red',
      textColor: 'blue',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})
