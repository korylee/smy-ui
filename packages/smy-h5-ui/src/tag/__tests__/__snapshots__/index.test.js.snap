// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render textColor correctly 1`] = `
"<div class=\\"smy-tag smy-tag--plain\\" style=\\"color: blue; background: rgb(255, 255, 255); border-color: red;\\">
  <!---->
</div>"
`;
